ec/ecresource/projectresource/delete_test.go

```diff
@@ -0,0 +1,115 @@
+package projectresource
+
+import (
+	"context"
+	"testing"
+
+	"github.com/elastic/terraform-provider-ec/ec/internal/gen/serverless/resource_elasticsearch_project"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestDelete(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	t.Run("should fail if reading the tf model errors", func(t *testing.T) {
+		ctx := context.Background()
+		req := resource.DeleteRequest{
```

Comment:
Technically, here and in the second test it can be replaced with `req := resource.DeleteRequest{}`

Reply:
Same as the create test. Using a non-zero value here does provide additional checks during test execution. I'm happy to remove that if we don't think it's worthwhile though.